compiler/rustc_const_eval/src/interpret/step.rs

```diff
@@ -199,12 +200,48 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             }
 
             Aggregate(box ref kind, ref operands) => {
-                assert!(matches!(kind, mir::AggregateKind::Array(..)));
-
-                for (field_index, operand) in operands.iter().enumerate() {
-                    let op = self.eval_operand(operand, None)?;
-                    let field_dest = self.place_field(&dest, field_index)?;
-                    self.copy_op(&op, &field_dest, /*allow_transmute*/ false)?;
+                self.write_uninit(&dest)?;
```

Comment:
This seems worth factoring into a separate function, e.g. `write_aggregate`, probably in `place.rs`.